python/mxnet/ndarray/ndarray.py

```diff
@@ -2479,6 +2479,8 @@ def array(source_array, ctx=None, dtype=None):
     if isinstance(source_array, NDArray):
         dtype = source_array.dtype if dtype is None else dtype
     else:
+        if isinstance(source_array, (float, int)):
```

Comment:
mx.base.numeric_types is better

Reply:
my concern here is that you implicitly converted 0-d array(constant) to 1-d array here, which may not be very appropriate

Reply:
Ah, my logic was since MXNet NDArray needs a 1d array minimum, I should convert constant. Else, is there a way I can do constant initialization support? Would be great to get some help in that direction.